Core/KSP.cs

```diff
@@ -440,6 +445,7 @@ public bool ScanGameData()
 
                 bool dlcChanged = manager.ScanDlc();
                 manager.Save(false);
+                manager.Dispose();
```

Comment:
@HebaruSan, is this safe to call here?
Problem was, `SetupCkanDirectories()` calls `ScanGameData()`,
which creates a new `RegistryManager.Instance()`, which creates a lock file.
This interferes with transactions.

Since this `RegistryManager` object is instantiated inside `ScanGameData()`, it can't be accessed after leaving the function scope, yes? This should make it safe to dispose it.

I tried to add a new KSP dir without a dir, which calls `ScanGameData()` too, and that still works fine :D

Reply:
> Since this RegistryManager object is instantiated inside ScanGameData(), it can't be accessed after leaving the function scope, yes?

No, any other code can access the same manager via `RegistryManager.Instance(ksp)` if it has a reference to the same `KSP ksp` object. In fact, it's possible that some other code *already* has such a reference and is depending on it not being disposed (GUI calls this function at the start of `UpdateRepo` and probably would prefer to keep its lock).

Reply:
Oh well, I see it's a bad idea.
Then I'll search another way to remove the lock file in failed transactions or prevent it from being created at all.

Reply:
I solved it with passing a bool to `KSP()` and `SetupCkanDirectories()` signaling whether `ScanGameData()` should be called. It's the easiest way I think.